File: python/Day25/time_complexity.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
import math

test_cases = int(input())
for _ in range(test_cases):
    num = int(input())
    
    if num == 1:
        print("Not prime")
        continue
    

    sqrt_num = int(math.sqrt(num))
    is_prime = True
    for i in range(2, sqrt_num+1):
        if num % i == 0:
            is_prime = False
            break
    
    if is_prime:
        print("Prime")
    else:
        print("Not prime")

File: cpp/Day25/time_complexity.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    int num_test_cases;
    cin >> num_test_cases;

    while(num_test_cases--){
        int num;
        cin >> num;

        if(num == 1){
            cout << "Not prime\n";
            continue;
        }

        bool is_prime = true;
        int sqrt_num = sqrt(num);
        for(int i = 2; i <= sqrt_num; i++){
            if(num % i == 0){
                is_prime = false;
                break;
            }
        }
        
        if(is_prime){
            cout << "Prime\n";
        }else{
            cout << "Not prime\n";
        }
    }
    return 0;
}
